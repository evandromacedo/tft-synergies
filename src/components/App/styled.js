import { createGlobalStyle } from 'styled-components';
import BeaufortRegular from '../../static/font/Beaufort-Regular.ttf';
import BeaufortBold from '../../static/font/Beaufort-Bold.ttf';

export const GlobalStyle = createGlobalStyle`
  /* Resets */
  *, *:before, *:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }

  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed,
  figure, figcaption, footer, header, hgroup,
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  body {
    -webkit-tap-highlight-color: transparent;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    line-height: 1;
  }

  ol, ul {
    list-style: none;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  input {
    outline-width: 0.3em;
    outline-color: transparent;
    -webkit-appearance: none;
    appearance: none;
    -webkit-border-radius:0;
    border-radius:0;
  }

  .input:disabled {
    background-color: transparent;
  }

  button {
    outline: none;
    cursor: pointer;
    background-color: transparent;
    margin: 0;
  }

  input,
  button {
    text-transform: inherit;
    font-family: inherit;
    letter-spacing: inherit;
    border: none;
  }

  /* Responsiveness */
  img,
  iframe,
  object,
  embed,
  video {
    height: auto;
    max-width: 100%;
  }

  /* Fonts */
  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortRegular}) format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Beaufort';
    src: url(${BeaufortBold}) format('truetype');
    font-weight: bold;
    font-style: normal;
  }

  html {
    font-family: 'Beaufort', sans-serif;
    font-weight: bold;
    font-size: 10px;
    color: #fff;
    background-color: #121212;
  }
`;
