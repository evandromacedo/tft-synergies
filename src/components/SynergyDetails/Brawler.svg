<?xml version="1.0" encoding="UTF-8"?>
<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 57.1 (83088) - https://sketch.com -->
    <title>Brawler</title>
    <desc>Created with Sketch.</desc>
    <g id="Brawler" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M14.3316129,19.5561032 C12.9845094,19.2740426 11.858069,18.8392057 10.9522581,18.2515795 C10.0464471,17.6639532 9.77935298,17.1899419 10.1509677,16.8295311 C10.2129035,16.766851 10.6464476,16.7472637 11.4516129,16.7707688 C12.2567782,16.7942738 13.193543,16.8530356 14.2619355,16.9470558 C16.0890414,17.1194261 17.1961271,17.2878764 17.5832258,17.4524118 C17.9703245,17.6169471 18.163871,17.9891048 18.163871,18.568896 C18.163871,19.3994078 17.9664536,19.8655843 17.5716129,19.9674395 C17.1767722,20.0692947 16.096783,19.932184 14.3316129,19.5561032 Z M2.13806452,13.2802864 C1.99870898,12.9198757 1.84774275,12.3361757 1.68516129,11.529169 C1.52257983,10.7221623 1.39483917,9.94258987 1.30193548,9.1904283 L1,6.44035129 L12.4735484,0 L19,2.04492906 L18.8374194,5.59417375 C18.7909675,6.56571578 18.7212908,7.67043652 18.6283871,8.90836912 C18.5354834,10.1463017 18.450323,11.1413339 18.3729032,11.8934954 L18.0941935,14.6905823 L16.2129032,14.5260478 C15.3458021,14.4476976 14.7845174,14.2831648 14.5290323,14.0324442 C14.2735471,13.7817237 14.006453,13.1079224 13.7277419,12.0110201 L13.1006452,9.66052693 L13.0541935,12.1755546 C13.0232257,13.3821472 12.9419361,14.095123 12.8103226,14.3145034 C12.678709,14.5338839 12.2722615,14.604398 11.5909677,14.5260478 C11.0025777,14.4633677 10.5845174,14.2753301 10.3367742,13.9619294 C10.089031,13.6485288 9.84129156,12.9982322 9.59354839,12.0110201 L8.96645161,9.66052693 L8.92,12.1285448 C8.8890321,13.3351373 8.80000074,14.0520305 8.65290323,14.279246 C8.50580572,14.5064615 8.06064888,14.6200675 7.31741935,14.6200675 C6.62064168,14.6200675 6.17161391,14.5103789 5.97032258,14.2909985 C5.76903125,14.071618 5.59096852,13.4840006 5.43612903,12.5281286 C5.20386981,11.0551455 4.96774314,10.2873254 4.72774194,10.2246453 C4.48774074,10.1619652 4.44516052,10.8436014 4.6,12.2695744 C4.72387159,13.3351366 4.7393553,13.9932681 4.64645161,14.2439886 C4.55354792,14.4947091 4.25161546,14.6200675 3.74064516,14.6200675 C3.38451435,14.6200675 3.06709817,14.5064615 2.7883871,14.279246 C2.50967603,14.0520305 2.292904,13.7190473 2.13806452,13.2802864 Z" id="Brawler-Copy" fill="#000000" fill-rule="nonzero"></path>
    </g>
</svg>