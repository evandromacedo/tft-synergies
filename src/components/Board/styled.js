import styled from 'styled-components';

export const Wrapper = styled.section`
  display: grid;
  grid-gap: 2.4rem;
`;

export const BoardChampions = styled.div`
  display: grid;
  grid-gap: 2.4rem;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
  justify-content: space-evenly;

  @media (max-width: 856px) {
    grid-template-columns: repeat(auto-fill, minmax(15rem, 1fr));
  }

  @media (max-width: ${props => props.theme.extraSmall}) {
    justify-items: center;
  }
`;

export const Text = styled.p`
  ${props => props.theme.sans16pxItalic}
  text-align: center;
  color: rgba(255, 255, 255, 0.8);
`;
