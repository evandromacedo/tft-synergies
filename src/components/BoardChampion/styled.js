import styled from 'styled-components';
import TierBorder from '../Common/TierBorder';

export const Wrapper = styled(TierBorder)`
  position: relative;
  height: 12.8rem;
  padding: 0.8rem;
  max-width: 25.6rem;
  display: grid;
  grid-gap: 0.8rem;

  /* Background image for Splash Art */
  &:before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    z-index: -1;
    background-image: url(${props => props.background});
    background-position: center right;
    background-size: cover;
  }
`;

export const Synergies = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-rows: repeat(3, 2rem);
`;

export const Items = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  grid-template-columns: 2.8rem 2.8rem 2.8rem 1fr;
  align-items: center;
`;

export const Item = styled.div`
  width: 2.8rem;
  height: 2.8rem;
  background-color: ${props => props.theme.background3};
`;

export const Name = styled.p`
  ${props => props.theme.bold20px}
  text-align: center;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
