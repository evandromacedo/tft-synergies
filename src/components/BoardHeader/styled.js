import styled from 'styled-components';

export const Wrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr auto;
  align-items: center;

  @media (max-width: ${props => props.theme.extraSmall}) {
    grid-template-columns: 1fr;
    grid-gap: 1.2rem;
    justify-items: center;
  }
`;
