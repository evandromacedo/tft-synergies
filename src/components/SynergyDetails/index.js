import React from 'react';
import PropTypes from 'prop-types';
import * as S from './styled';

export default function SynergyDetails({ details, synergyIndex }) {
  return (
    <S.Wrapper>
      <S.Header>
        <SvgBrawler />
        <S.Name>{details.name}</S.Name>
      </S.Header>
      <S.Description>
        {details.description || 'No description provided at the moment.'}
      </S.Description>
      <S.SynergyList>
        {details.bonuses.map((bonus, index) => (
          <S.SynergyItem key={index} className={synergyIndex === index ? 'active' : ''}>
            ({bonus.needed}) {bonus.effect}
          </S.SynergyItem>
        ))}
      </S.SynergyList>
    </S.Wrapper>
  );
}

SynergyDetails.propTypes = {
  details: PropTypes.object.isRequired,
  synergyIndex: PropTypes.number
};

// This is just a mock and will me replaced when more icons are added
const SvgBrawler = () => (
  <svg width={20} height={20}>
    <path
      d="M14.332 19.556c-1.347-.282-2.474-.717-3.38-1.304-.906-.588-1.173-1.062-.801-1.422.062-.063.495-.083 1.3-.06a48.8 48.8 0 012.81.177c1.828.172 2.935.34 3.322.505.387.165.58.537.58 1.117 0 .83-.197 1.297-.591 1.398-.395.102-1.475-.035-3.24-.41zM2.138 13.28c-.14-.36-.29-.944-.453-1.75a31.257 31.257 0 01-.383-2.34L1 6.44 12.474 0 19 2.045l-.163 3.55c-.046.97-.116 2.075-.209 3.313a113.404 113.404 0 01-.255 2.985l-.279 2.798-1.881-.165c-.867-.078-1.428-.243-1.684-.494-.255-.25-.523-.924-.801-2.021l-.627-2.35-.047 2.515c-.03 1.206-.112 1.92-.244 2.139-.131.219-.538.29-1.219.211-.588-.063-1.006-.25-1.254-.564-.248-.313-.496-.964-.743-1.951l-.628-2.35-.046 2.468c-.031 1.206-.12 1.923-.267 2.15-.147.227-.592.341-1.336.341-.696 0-1.145-.11-1.347-.329-.201-.22-.379-.807-.534-1.763-.232-1.473-.468-2.24-.708-2.303-.24-.063-.283.619-.128 2.045.124 1.065.14 1.723.046 1.974-.092.25-.394.376-.905.376-.356 0-.674-.114-.953-.34-.278-.228-.495-.561-.65-1z"
      fill="#D8A75B"
      fillRule="nonzero"
    />
  </svg>
);
