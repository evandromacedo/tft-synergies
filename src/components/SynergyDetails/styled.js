import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 28rem;
  padding: 1.2rem;
  border: 1px solid ${props => props.theme.accentYellow};
  background-color: rgba(0, 0, 0, 0.8);
  position: relative;
  z-index: 2;
`;

export const Header = styled.div`
  border-bottom: 1px solid ${props => props.theme.accentYellow};
  padding-bottom: 8px;
  display: grid;
  grid-gap: 0.8rem;
  grid-template-columns: 2rem 1fr;
  align-items: center;
  margin-bottom: 0.8rem;
`;

export const Name = styled.h1`
  transform: translateY(2px);
  ${props => props.theme.bold20px};
`;

export const Description = styled.p`
  ${props => props.theme.sans16px};
  margin-bottom: 0.8rem;
`;

export const SynergyList = styled.ul`
  display: grid;
  grid-gap: 0.4rem;
`;

export const SynergyItem = styled.li`
  ${props => props.theme.sans16px};
  color: rgba(255, 255, 255, 0.4);

  &.active {
    color: white;
  }
`;
