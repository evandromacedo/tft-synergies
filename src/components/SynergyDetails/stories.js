import React from 'react';
import SynergyDetails from '.';
import { storiesOf } from '@storybook/react';

storiesOf('SynergyDetails', module).add('Default', () => <SynergyDetails />);
