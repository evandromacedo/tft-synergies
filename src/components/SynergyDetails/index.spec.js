import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as S from './styled';
import SynergyDetails from '.';

const demonDetailsMock = {
  key: 'demon',
  name: 'Demon',
  description:
    'Demons basic attacks have a 40% chance to burn 20 mana from their target and return mana to the attacker.',
  accentChampionImage: 'https://cdn.blitz.gg/blitz/centered/Aatrox_Splash_Centered_0.jpg',
  bonuses: [
    { needed: 2, effect: 'Return 15 mana back to attacker' },
    { needed: 4, effect: 'Return 30 mana back to attacker' },
    { needed: 6, effect: 'Return 45 mana back to attacker' }
  ]
};

describe('<SynergyDetails />', () => {
  it('renders properly', () => {
    shallow(<SynergyDetails details={demonDetailsMock} />);
  });

  it('renders name properly', () => {
    const wrapper = shallow(<SynergyDetails details={demonDetailsMock} />);
    const Name = wrapper.find(S.Name);
    expect(Name.text()).toEqual('Demon');
  });

  it('renders description of details', () => {
    const wrapper = shallow(<SynergyDetails details={demonDetailsMock} />);
    const DescriptionDemon = wrapper.find(S.Description);
    expect(DescriptionDemon.text()).toEqual(demonDetailsMock.description);
  });

  it('renders "no description provided" when description is null', () => {
    const wrapper = shallow(
      <SynergyDetails details={{ ...demonDetailsMock, description: null }} />
    );
    const DescriptionNull = wrapper.find(S.Description);
    expect(DescriptionNull.text()).toEqual('No description provided at the moment.');
  });

  it('render SynergyList with bonuses', () => {
    const wrapper = shallow(
      <SynergyDetails synergyIndex={1} details={demonDetailsMock} />
    );
    const SynergyList = wrapper.find(S.SynergyList);
    expect(toJSON(SynergyList)).toMatchSnapshot();
  });
});
