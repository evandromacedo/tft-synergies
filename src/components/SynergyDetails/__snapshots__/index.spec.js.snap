// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SynergyDetails /> render SynergyList with bonuses 1`] = `
<styled.ul>
  <styled.li
    className=""
    key="0"
  >
    (
    2
    ) 
    Return 15 mana back to attacker
  </styled.li>
  <styled.li
    className="active"
    key="1"
  >
    (
    4
    ) 
    Return 30 mana back to attacker
  </styled.li>
  <styled.li
    className=""
    key="2"
  >
    (
    6
    ) 
    Return 45 mana back to attacker
  </styled.li>
</styled.ul>
`;
